app/controllers/api/v1/ability.rb

```diff
@@ -96,7 +97,12 @@ def beneficiary_abilities
       end
 
       def canned_response_abilities
-        can %I[index create update destroy show], CannedResponse if can_manage_canned_response?
+        can %i[index create update destroy show], CannedResponse if can_manage_canned_response?
+      end
+
+      def offer_response_abilities
+        can [:create, :index], OfferResponse, user_id: @user.id
```

Comment:
This permission rule would still allow me to create an `offer_response` record to an offer that isn't shared

Reply:
(do write a test for that)

Reply:
sure , working on it

Reply:
Yes, `user_id` is correct here but we need to scope `offer_id` also. It should be the list of offers that the user is has created.

```
can [:create, :index], OfferResponse, { user_id: @user_id, offer: { created_by_id: @user.id } }
```

Reply:
@steveyken I think you're thinking of a different rule here. I think here `offer_id` is scoped to the shared offers

i.e

```ruby
can [:create, :index], OfferResponse, { user_id: @user_id } do |offer_response|
  Shareable.non_expired.find_by(
    resource_type: 'Offer',
    resource_id: offer_response.offer_id
  ).present?
end
```